src/modules/pythonmonkey/pythonmonkey.cc

```diff
@@ -22,8 +24,65 @@
 #include <datetime.h>
 
 typedef std::unordered_map<PyType *, std::vector<JS::PersistentRooted<JS::Value> *>>::iterator PyToGCIterator;
+typedef struct {
+  PyObject_HEAD
+} NullObject;
+
 std::unordered_map<PyType *, std::vector<JS::PersistentRooted<JS::Value> *>> PyTypeToGCThing; /**< data structure to hold memoized PyObject & GCThing data for handling GC*/
 
+// @TODO (Caleb Aikens) figure out how to use C99-style designated initializers with a modern C++ compiler
+static PyTypeObject NullType = {
+  .ob_base = PyVarObject_HEAD_INIT(NULL, 0)
+  .tp_name = "pythonmonkey.Null",
+  .tp_basicsize = sizeof(NullObject),
+  .tp_itemsize = 0,
+  .tp_dealloc = NULL,
+  .tp_vectorcall_offset = NULL,
+  .tp_getattr = NULL,
+  .tp_setattr = NULL,
+  .tp_as_async = NULL,
+  .tp_repr = NULL,
+  .tp_as_number = NULL,
+  .tp_as_sequence = NULL,
+  .tp_as_mapping = NULL,
+  .tp_hash = NULL,
+  .tp_call = NULL,
+  .tp_str = NULL,
+  .tp_getattro = NULL,
+  .tp_setattro = NULL,
+  .tp_as_buffer = NULL,
+  .tp_flags = Py_TPFLAGS_DEFAULT,
+  .tp_doc = PyDoc_STR("Javascript null object"),
+  .tp_traverse = NULL,
+  .tp_clear = NULL,
+  .tp_richcompare = NULL,
+  .tp_weaklistoffset = NULL,
+  .tp_iter = NULL,
+  .tp_iternext = NULL,
+  .tp_methods = NULL,
+  .tp_members = NULL,
+  .tp_getset = NULL,
+  .tp_base = NULL,
+  .tp_dict = NULL,
+  .tp_descr_get = NULL,
+  .tp_descr_set = NULL,
+  .tp_dictoffset = NULL,
+  .tp_init = NULL,
+  .tp_alloc = NULL,
+  .tp_new = PyType_GenericNew,
+  .tp_free = NULL,
+  .tp_is_gc = NULL,
+  .tp_bases = NULL,
+  .tp_mro = NULL,
+  .tp_cache = NULL,
+  .tp_subclasses = NULL,
+  .tp_weaklist = NULL,
+  .tp_del = NULL,
+  .tp_version_tag = NULL,
+  .tp_finalize = NULL,
+  .tp_vectorcall = NULL,
+};
+
 static void cleanup() {
   JS_DestroyContext(cx);
```

Comment:
```suggestion
  if (cx) JS_DestroyContext(cx);
```

`cx` may be a nullptr if `JS_NewContext` fails